evalml/automl/automl_search.py

```diff
@@ -383,6 +383,8 @@ def __init__(self,
 
         if self.allowed_pipelines == []:
             raise ValueError("No allowed pipelines to search")
+
+        logger.info(f"{len(self.allowed_pipelines)} pipelines ready for search.")
```

Comment:
nit, but can we add simple test coverage for this just to make sure the output is expected for diff numbers of pipelines/iterations/model families?